File: Solutions_Easy/0242_ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        s_list = [0] * 26
        t_list = [0] * 26
        for sc, tc in zip(s, t):
            s_list[ord(sc) - 97] += 1
            t_list[ord(tc) - 97] += 1

        for so, to in zip(s_list, t_list):
            if so != to:
                return False

        return True

File: Solutions_Easy/0242_ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        std::array<int, 26> counts{};
        for (size_t i = 0; i < s.size(); ++i) {
            ++counts[s[i] - 'a'];
            --counts[t[i] - 'a'];
        }
        for (int c : counts) {
            if (c != 0) {
                return false;
            }
        }
        return true;
    }
};